cmd/crc-embedder/cmd/embed.go

```diff
@@ -124,9 +143,25 @@ func getEmbedFileList(goos string, destDir string) []string {
 	return fileList
 }
 
-func downloadDataFiles(goos string, destDir string) ([]string, error) {
+func downloadDataFiles(goos string, components []string, destDir string) ([]string, error) {
 	downloadedFiles := []string{}
 	downloads := dataFileUrls[goos]
+
+	if len(components) > 0 {
+		for _, component := range components {
+			if dl, ok := downloads[component]; ok {
+				filename, err := download.Download(dl.url, destDir, dl.permissions, nil)
+				if err != nil {
+					return nil, err
+				}
+				downloadedFiles = append(downloadedFiles, filename)
+			} else {
+				logging.Warnf("'%s' is not available, ensure goos flag is used. Supported component names are: %s", component, strings.Join(componentNames, ", "))
+			}
+		}
+		return downloadedFiles, nil
+	}
+
 	for _, dl := range downloads {
```

Comment:
doesn't `downloads` now is the map instead of slice so you need to have `dl, ok` logic here.

Reply:
`for i, dl := range slice` will return index in the array  + array element, `for key, dl := range map` will return map key and map value. `for _, dl := range map` and `for _dl := range slice` should have the same behaviour (but it also got me thinking for a bit)